packages/block-library/src/search/edit.js

```diff
@@ -148,16 +149,21 @@ export default function SearchEdit( {
 				) }
 
 				{ ! buttonUseIcon && (
-					<RichText
-						className="wp-block-search__button"
-						aria-label={ __( 'Button text' ) }
-						placeholder={ __( 'Add button text…' ) }
-						withoutInteractiveFormatting
-						value={ buttonText }
-						onChange={ ( html ) =>
-							setAttributes( { buttonText: html } )
-						}
-					/>
+					// This rule only makes sense for UI not blocks.
+					// eslint-disable-next-line react/forbid-elements
+					<button type="submit" className="wp-block-search__button">
+						<RichText
```

Comment:
It seems there's something that prevents RichText from working as a "button". I've used a small hack here to make it work and still keep a similar markup to the frontend.

Reply:
What prevents it from working? Do you get any errors?

Reply:
It's just stuck, like you can't edit it. Maybe "contenteditable" doesn't work on buttons?

Reply:
That's interesting. I'll try this out in a bit.

Reply:
It should work: https://codepen.io/iseulde/pen/vYgOpVw

Maybe something is wrong in RichText.

Reply:
For some reason the selection change event is not firing in this context. It is in codepen though. Strange.

Reply:
It seems something is the block editor is causing the issue. I insert `<button contenteditable>test</button>` without rich text in a block and it also fails to set selection. The same line on an empty web page works fine.

Reply:
@ellatrix after the removal of revert: all, I have two bugs:

 - In safari, backspace removes the block and not a character in the text
 - in chrome, I can't type an empty space

Reply:
Hm, yes, Space will conflict with "pressing" the button.
I'll have a look at both issues. In the meantime, feel free to nest it in a span and we can adjust later.